### DISCUSSION TITLE
Getting 404 inserting into a custom schema but works fine in public

### DISCUSSION THREAD

**Discussion by brysonkbarney** [supabase/supabase#31299] [Questions] [answered]
Hey all, I’m getting a 404 when doing POST /rest/v1/rinse.customers. If I insert into public.test_table via POST /rest/v1/test_table, I get 201 no problem. I’ve:

- Exposed rinse in Project Settings → API
- Granted all permissions as directed in the docs
- Verified spelling of rinse.customers
- Used the service role key (which works for public)

Any ideas what could be causing the 404 on my custom schema? Thanks!

Here is my code for reference:

```
SUPABASE_URL = os.getenv("SUPABASE_URL")
SUPABASE_SERVICE_ROLE_KEY = os.getenv("SUPABASE_SERVICE_ROLE_KEY")

url = f"{SUPABASE_URL}/rest/v1/rinse.test_rinse_table"
headers = {
    "apikey": SUPABASE_SERVICE_ROLE_KEY,
    "Authorization": f"Bearer {SUPABASE_SERVICE_ROLE_KEY}",
    "Content-Type": "application/json",
    "Prefer": "return=representation" 
}

data = {"note": "Hello Rinse schema!"}

resp = requests.post(url, headers=headers, data=json.dumps(data))
print("Status Code:", resp.status_code) 
```

**Comment by GaryAustin1** [comment, marked as answer]
Using non default schema of public requires 1 of 2 headers be set with the schema.  You can't do schema.table.
https://docs.postgrest.org/en/v12/references/api/schemas.html#schemas
